fastlane_bot/helpers/txhelpers.py

```diff
@@ -969,13 +972,57 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
                             base_gas_price=baseFee, max_priority_fee=max_priority, nonce=self.get_nonce()
                         )
                     )
+                    approve_tx = self.sign_transaction(approve_tx)
+                    self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
+                    return self.submit_transaction(approve_tx)
                 except Exception as e:
                     self.ConfigObj.logger.info(
                         f"(***2***) Error when building transaction: {e.__class__.__name__} {e}")
             else:
                 return None
-        self.ConfigObj.logger.info(f"Submitting approval for token: {token_address}")
-        return self.submit_transaction(approve_tx)
 
+    async def get_layer_one_gas_fee(self, signed_transaction) -> Decimal:
+        """
+        Returns the expected layer one gas fee for a layer 2 Optimism transaction
+        :param signed_transaction: the signed ethereum TX
+
+        returns: Decimal
+            The total fee (in gas token) for the l1 gas fee
+        """
+        ethereum_base_fee = await self._get_layer_one_gas_price()
+        fixed_overhead = await self._get_layer_one_fee_overhead()
+        dynamic_overhead = await self._get_layer_one_fee_scalar()
```

Comment:
Consider running these async calls outside of this context (e.g., at the beginning of the loop), then you do not need to introduce any asynchronous-ity as part of this context (which seems to be forcing you to execute `run_until_complete` as part of this context).

Reply:
Well, the suggestion here is not just to add them, but rather - to add them only when needed.

Reply:
Where could we add this that would be more efficient? 

Currently these are only being called when needed.

Reply:
I shall look into it...